crates/holochain/src/core/ribosome/real_ribosome.rs

```diff
@@ -732,105 +785,88 @@ impl RealRibosome {
+    ) -> Result<ExternIO, RibosomeError> {
+        let fn_name = fn_name.clone();
+        let result: Result<ExternIO, RuntimeError>;
+        {
+            let instance = instance_with_store.instance.clone();
+            let mut store_lock = instance_with_store.store.lock();
+            let mut store_mut = store_lock.as_store_mut();
+            result = holochain_wasmer_host::guest::call(
+                &mut store_mut,
+                instance,
+                fn_name.as_ref(),
+                // be aware of this clone!
+                // the whole invocation is cloned!
+                // @todo - is this a problem for large payloads like entries?
+                invocation.to_owned().host_input()?,
+            );
+        }
 
-            // a bit of typefu to avoid cloning the result.
-            let (can_cache, result) = match result {
-                Err(runtime_error) => {
-                    // This will bubble up and be logged later but capture zome/function that was called while the context is available
-                    tracing::error!(?runtime_error, ?zome, ?to_call);
-                    match runtime_error.downcast::<WasmError>() {
-                        Ok(wasm_error) => {
-                            (!wasm_error.error.maybe_corrupt(), Err(wasm_error.into()))
-                        }
-                        Err(result) => (false, Err(result)),
-                    }
+        // a bit of typefu to avoid cloning the result.
+        let (can_cache, result) = match result {
+            Err(runtime_error) => {
+                // This will bubble up and be logged later but capture zome/function that was called while the context is available
+                tracing::error!(?runtime_error, ?zome, ?fn_name);
+                match runtime_error.downcast::<WasmError>() {
+                    Ok(wasm_error) => (!wasm_error.error.maybe_corrupt(), Err(wasm_error.into())),
+                    Err(result) => (false, Err(result)),
                 }
-                result => (true, result),
-            };
-
-            // Cache this instance.
-            if can_cache {
-                self.cache_instance(context_key, instance_with_store, zome.zome_name())?;
             }
+            result => (true, result),
+        };
 
-            Ok(Some(result?))
-        } else {
-            // the func doesn't exist
-            // the callback is not implemented
-            Ok(None)
+        // Cache this instance.
+        tracing::error!("can cache {can_cache}");
+        if can_cache {
+            self.cache_instance(context_key, instance_with_store, zome.zome_name())?;
         }
+
+        Ok(result?)
     }
 
     pub fn get_const_fn_for_wasm(
         &self,
         call_context: CallContext,
         name: &str,
-        module: Arc<Module>,
     ) -> Result<Option<i32>, RibosomeError> {
-        // Check if the wasm has a function that matches this type.
-        if module.exports().functions().any(|f| {
```

Comment:
Same as above, this is checked in the calling function and module is not needed here.